$NetBSD: patch-ae,v 1.6 2009/07/27 12:59:44 obache Exp $

Accept ENOTCONN as indicating socket is not ready for write.

--- src/core/network.c.orig	2005-11-19 23:45:08.000000000 +0100
+++ src/core/network.c
@@ -373,7 +373,7 @@ int net_transmit(GIOChannel *handle, con
 
 	err = g_io_channel_write(handle, (char *) data, len, &ret);
 	if (err == G_IO_ERROR_AGAIN ||
-	    (err != 0 && (errno == EINTR || errno == EPIPE)))
+	    (err != 0 && (errno == EINTR || errno == EPIPE || errno == ENOTCONN)))
 		return 0;
 
 	return err == 0 ? (int)ret : -1;
